lib/fluent/plugin/geoip.rb

```diff
@@ -116,35 +129,51 @@ def parse_json(message)
     def get_address(record)
       address = {}
       @geoip_lookup_key.each do |field|
-        address.store(field, record[field]); next if not record[field].nil?
-        key = field.split('.')
-        obj = record
-        key.each {|k|
-          break obj = nil if not obj.has_key?(k)
-          obj = obj[k]
-        }
-        address.store(field, obj)
+        address[field] = record[field] || record.dig(*field.split('.'))
```

Comment:
Right. But I've added dependency to [dig_rb gem](https://github.com/jrochkind/dig_rb) to use `dig` method.